pallets/avn-transaction-payment/src/fee_adjustment_config.rs

```diff
@@ -176,6 +187,37 @@ impl<T: Config> TimeBasedConfig<T> {
     }
 }
 
+#[derive(Encode, Decode, MaxEncodedLen, Default, Clone, PartialEq, Debug, Eq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct TransactionBasedConfig<T: Config> {
+    pub fee_type: FeeType<T>,
+    account: T::AccountId,
+    end_count: T::Index,
+}
+
+impl<T: Config> TransactionBasedConfig<T> {
+    pub fn is_valid(&self) -> bool {
+        return !self.end_count.is_zero() && self.fee_type != FeeType::None
+    }
+
+    pub fn is_active(&self) -> bool {
+        return self.end_count >= <frame_system::Pallet<T>>::account(&self.account).nonce
+    }
+
+    pub fn get_fee(&self, original_fee: BalanceOf<T>) -> Result<BalanceOf<T>, Error<T>> {
+        if self.is_active() {
+            return calculate_fee::<T>(original_fee, &self.fee_type)
+        }
+
+        // There is no adjutment to make so return the original fee
```

Comment:
```suggestion
        // There is no adjustment to make so return the original fee
```
and possibly in other places